dev/io.openliberty.microprofile.telemetry.1.0.internal_fat/fat/src/io/openliberty/microprofile/telemetry/internal_fat/apps/rest/rest/client/user/MessageAndRestClientTestServlet.java

```diff
@@ -0,0 +1,145 @@
+        assertTrue(results.contains("LENGTH 12!!!"));
+    }
+
+    @Test
+    public void testSpanInBothJaxCallerAndCalled() throws InterruptedException {
+        getResultsWithWaitIfNeeded();//we just want to delay until the results are in.
+        List<List<SpanData>> listOfListOfResults = new ArrayList<List<SpanData>>(5);
+
+        SpanDataMatcher spanInsideClientMatcher = SpanDataMatcher.hasName("SimpleReactiveMessagingBean.toUpperCase");
+
+        SpanDataMatcher spanInsideJaxServiceMatcher = SpanDataMatcher.hasName("RestClientThatCapitalizes.jaxServiceThatCapitalizes");
+
+        //Each message will end up with four spans being registered. And there are five messages.
+        for (Span s : spanAndExpectedNumber.keySet()) {
+            listOfListOfResults.add(new ArrayList(inMemorySpanExporter.getFinishedSpanItems(spanAndExpectedNumber.get(s), s)));
+        }
+
+        String failMessage = "";
+
+        //We don't really need to test this five times, but keeping all five
+        //makes the test app more authentic and since we have them.
+        for (int index = 0; index < listOfListOfResults.size(); index++) {
+            boolean foundSpanCallingJax = false;
+            boolean foundSpanInsideJax = false;
+
+            for (SpanData sd : listOfListOfResults.get(index)) {
+                if (spanInsideClientMatcher.matches(sd)) {
+                    foundSpanCallingJax = true;
+                }
+
+                if (spanInsideJaxServiceMatcher.matches(sd)) {
+                    foundSpanInsideJax = true;
+                }
+
+                if (foundSpanCallingJax && foundSpanInsideJax) {
+                    break;
+                }
+            }
+            if (!foundSpanCallingJax || !foundSpanInsideJax) {
+                StringBuilder sb = new StringBuilder();
+                sb.append("failed on index: " + index);
+                if (!foundSpanCallingJax) {
+                    sb.append(" Did not find a Span from SimpleReactiveMessagingBean");
+                }
+                if (!foundSpanInsideJax) {
+                    sb.append(" Did not find a Span from RestClientThatCapitalizes");
+                }
+                sb.append(System.lineSeparator());
+                failMessage = failMessage + sb.toString();
+            }
+        }
+
+        if (!failMessage.isEmpty()) {
+            Assert.fail(failMessage);
+        }
+    }
+
+    private List<String> getResultsWithWaitIfNeeded() throws InterruptedException {
+        List<String> results = reactiveBean.getResults();
+
+        if (results.size() < 3) {
+            return results;
+        }
+
+        long maxNanos = System.nanoTime() + millisToNanos(MAX_WAIT);
+        //Reactive messaging will populate this on another thread
+        synchronized (results) {
+            while (results.size() < 3 && nanoTimeRemaining(maxNanos)) {
+                results.wait(WAIT_INTERVAL);
+            }
+        }
+        return results;
+    }
```

Comment:
If the wait times out, this method should throw an exception rather than returning the results it has.